// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "datetime.proto";

option go_package = "github.com/Funkit/satellite-data-provider/pointing";
option java_multiple_files = true;
option java_package = "funkit.satellite-data-provider";
option java_outer_classname = "AntennaPointingProto";

package pointing;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc GetAntennaPointing (AntennaPointingRequest) returns (AntennaPointingReply) {}
}

// The request message containing the user's name.
message AntennaPointingRequest { 
  SatelliteInformation satellite_information = 1;
  GroundStationInformation ground_station_information = 2;
}

message SatelliteInformation {
  string satellite_name = 1;
  string tle_line_1 = 2;
  string tle_line_2 = 3;
}

message GroundStationInformation {
  double station_latitude = 1;
  double station_longitude = 2;
  double station_altitude = 3;
  google.type.DateTime start_date = 4;
  google.type.DateTime stop_date = 5;
}

// The response message containing the greetings
message AntennaPointingReply {
  string message = 1;
}
